python/GafferUI/ColorChooser.py

```diff
@@ -235,7 +235,8 @@ def __componentValueChanged( self, componentWidget, reason ) :
 			a = { "r" : 0, "g" : 1, "b" : 2, "a" : 3 }[componentWidget.component]
 			newColor[a] = componentValue
 		else :
-			newColor = newColor.rgb2hsv()
+			newColor = self.__colorHsv.__class__( self.__colorHsv )
+
 			a = { "h" : 0, "s" : 1, "v" : 2 }[componentWidget.component]
 			newColor[a] = componentValue
 			newColor = newColor.hsv2rgb()
```

Comment:
Yes, I see that now, that's a very good point.

I think I have a good solution for that in https://github.com/GafferHQ/gaffer/pull/5909/commits/54983339d69c982406c4d1cdcab49c340a2ca09c. It's a fairly small change - I considered adding `__setColorInternalHSV()` and `__setColorInternalRGB()` methods instead of the new argument, but that seemed like a more complicated solution than adding this enum argument.